functions/controllers/VerifyRecaptcha/index.js

```diff
@@ -0,0 +1,65 @@
+const functions = require("firebase-functions");
+const express = require("express");
+const cors = require("cors");
+const { base_google_endpoint, secretKey } = require("../../utils/config");
+const { corsConfig } = require("../../utils/config");
+const fetch = require("isomorphic-fetch");
+
+const verifyRecaptcha = express();
+const corsOptions = {
+  origin: corsConfig,
+  optionsSuccessStatus: 200,
+};
+verifyRecaptcha.use(cors(corsOptions));
+verifyRecaptcha.use(express.json());
+
+verifyRecaptcha.post("/submit", async (req, res) => {
+  // Logic to process case of no g-recaptcha-response
+  functions.logger.log(req.body);
+  if (Object.keys(req.body).includes("captcha")) {
+    if (req.body.captcha === undefined || req.body.captcha === "" || req.body.captcha === null) {
+      return res.status(401).send({ error: true, status: 401, message: "Please select captcha" });
+    }
+    const formBody = `secret=${secretKey}&response=${req.body.captcha}&remoteip=${req.socket.remoteAddress}`;
```

Comment:
Can we grab the captcha key from the headers instead. The main reason is when requests are sent only headers are encrypted and so if a request is caught from the client-side people would be able to see what the key is since the body is not encrypted.

Reply:
good catch, pushing updated version now